newIDE/app/src/MainFrame/index.js

```diff
@@ -96,6 +96,8 @@ const GD_STARTUP_TIMES = global.GD_STARTUP_TIMES || [];
 
 const gd = global.gd;
 
+// const app = require('electron') // Doing this crashes the app.
```

Comment:
nit: remove commented out code

Reply:
Look at other files to see how to properly require electron... though not sure we want to do it in this file at all.

Reply:
I think hes trying to trigger CloseConfirmDialog  to appear. He just needs to put

Reply:
Thanks for the review. I'll find the right way.

Reply:
> I think hes trying to trigger CloseConfirmDialog to appear, but currently the prop that calls it is !!this.state.currentProject

Not exactly ;) The props is there to tell if the dialog should be shown or not *when* the window is trying to be closed.
CloseConfirmDialog has its logic coded inside to show itself or not.

So finding a way to "trigger" the closing (only works for Electron, you can not do this in a web-app) will make the dialog automatically kicks in :)

Reply:
I understand that I will trigger the closing somehow then CloseConfirmDialog will kick in. I am just trying to imitate other functions in the index.js and created an ipcRenderer listening the event. I think if this.state.currentProject is true the quit box should be popped. (because shouldPrompt will be true in line 32 in CloseConfirmDialog.js) I am trying to solve the problem, will check other files.